openpype/hosts/flame/api/lib.py

```diff
@@ -551,11 +559,119 @@ def get_segment_attributes(segment):
         attr = getattr(segment, attr_name)
         segment_attrs_data[attr] = str(attr).replace("+", ":")
 
-        if attr in ["record_in", "record_out"]:
+        if attr_name in ["record_in", "record_out"]:
             clip_data[attr_name] = attr.relative_frame
         else:
             clip_data[attr_name] = attr.frame
 
     clip_data["segment_timecodes"] = segment_attrs_data
 
     return clip_data
+
+
+def get_clips_in_reels(project):
+    output_clips = []
+    project_desktop = project.current_workspace.desktop
+
+    for reel_group in project_desktop.reel_groups:
+        for reel in reel_group.reels:
+            for clip in reel.clips:
+                clip_data = {
+                    "PyClip": clip,
+                    "fps": float(str(clip.frame_rate)[:-4])
+                }
+
+                attrs = [
+                    "name", "width", "height",
+                    "ratio", "sample_rate", "bit_depth"
+                ]
+
+                for attr in attrs:
+                    val = getattr(clip, attr)
+                    clip_data[attr] = val
+
+                version = clip.versions[-1]
+                track = version.tracks[-1]
+                for segment in track.segments:
+                    segment_data = get_segment_attributes(segment)
+                    clip_data.update(segment_data)
+
+                output_clips.append(clip_data)
+
+    return output_clips
+
+
+def get_reformated_path(path, padded=True):
+    """
+    Return fixed python expression path
+
+    Args:
+        path (str): path url or simple file name
+
+    Returns:
+        type: string with reformated path
+
+    Example:
+        get_reformated_path("plate.1001.exr") > plate.%04d.exr
+
+    """
+    padding = get_padding_from_path(path)
+    found = get_frame_from_path(path)
+
+    if not found:
+        log.info("Path is not sequence: {}".format(path))
+        return path
+
+    if padded:
```

Comment:
This could cause an issues in future. E.g. if frame is `1001` and path is `F:\my_project\se01\sh1001\...\renderMain_sh1001.1001.exr` the result would be `F:\my_project\se01\sh%d\...\renderMain_sh%d.%d.exr`.

Reply:
Not sure why, if you add basename to the path var it should only change basename. on line 670 there is regex `[._](\d+)[.]` which should only match to .[_]1001. so sh1001 will not be matching. Am I right?

But yeah the `path` variable is perhaps misleading. `fname` or `basename` could be working better

Reply:
> Am I right?

Your are right. But the function will return string `"1001"` and all matching parts in filename are replaced here.

```
import re

filename = "renderMain_sh1001.1001.exr"

frame_pattern = re.compile(r"[._](\d+)[.]")
found = re.findall(frame_pattern, filename)
replacement = found.pop()

result = filename.replace(replacement, "%d")
print(result)

>>> renderMain_sh%d.%d.exr
```

Reply:
I think the regex should be `[\._](\d+)[\.]` because `.` is "any character".

BTW this way it will always replace only frame matching the frame regex:
```
def replace_frame_with(filename, replacement):
    frame_pattern = re.compile(r"[\._](\d+)[\.]")
    match = frame_pattern.search(filename)
    if not match:
        return filename

    start_idx, end_idx = match.span(1)

    return replacement.join(
        [filename[:start_idx], filename[end_idx:]]
    )
```

Reply:
I see your point, thanks for suggestion ;)